src/System.Text.RegularExpressions/src/System/Text/RegularExpressions/RegexCollectionDebuggerProxy.cs

```diff
@@ -0,0 +1,32 @@
+﻿// Copyright (c) Microsoft. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+using System.Collections.Generic;
+using System.Diagnostics;
+
+namespace System.Text.RegularExpressions
+{
+    internal sealed class RegexCollectionDebuggerProxy<T>
```

Comment:
@KrzysztofCwalina / @terrajobst , lots of assemblies end up having a debugger proxy like this.  Have we considered adding public ```CollectionDebuggerProxy<T>``` for any ```ICollection<T>``` implementation to System.Collections.dll?  Then any assembly implementing an ```ICollection<T>``` could just use that one public implementation rather than crafting their own and effectively duplicating the same implementation over and over.